### ISSUE TITLE
TabBarNavigator not rendering screen

### ISSUE THREAD

**Issue by Himujjal** [react-community/react-navigation#2434]
I am not able to render TabBarNavigator screen. But when I pass another functional component just for testing purposes it works absolutely fine.

Here's my first code: 
``` import React, {Component} from 'react'; 
import {View,Text} from 'react-native';
import { StackNavigator, TabNavigator } from 'react-navigation';
import PeopleList from './PeopleList';
import CompanyList from './CompanyList';
import AddPerson from './AddPerson';

const B = () => (<View><Text>This is Navigation B</Text></View>);
const C = () => (<View><Text>This is Navigation C </Text></View>);
const D = () => (<View><Text>This is Navigation D</Text></View>);

const Navigation = TabNavigator({
    PeopleList: { screen: B },
    AddPerson: { screen: C },
    CompanyList: { screen: D },
}, {
    tabBarOptions: {
        activeTintColor: 'white',
        inactiveTintColor: '#80cbc4',
        showLabel: false,
        swipeEnabled: false,
        style: {
            backgroundColor: '#26a69a',
        },
        lazy: true
    },
});

const A = () => (<View><Text>This is Navigation A</Text></View>);

export default Navigation;  
```

So when I am trying to render Navigation nothing (blank white screen) shows up.
Then again if I do this:

`export default A; `

`This is Navigation A` shows up in the screen.

Please run it in your emulator. I am using Android Nexus-4-API_23/Windows 10.

Things to note here: 
1. I am using Redux here but I am not in any sort dispatching the navigation state to the store. I don't know if that is compulsory but I don't think this should be an issue.
2. I am importing the `<Navigation/>` component to another screen
2. My dependencies are: 
```
"react": "15.4.2",
"react-native": "0.42.3",
"react-navigation": "^1.0.0-beta.7",
"react-native-material-kit": "^0.4.1",
"react-native-vector-icons": "^4.0.0",
"react-redux": "^5.0.3",
"redux": "^3.6.0",
"redux-thunk": "^2.2.0"
```
Oh! Yeah, Material-Kit only works with 0.42.3 version of react-native only. Please tell me if its a version problem.

**Comment by tudor2it** [comment]
@Himujjal Hi! Please let us know how did you solve the issue?

**Comment by Himujjal** [comment]
Actually I was having the TabNavigator wrapped inside a `<View>` element in the module where it is imported. Like this:
```
<View> 
     <Navigation/>
</View>
```
I did not realise that the navigating routed should not be wrapped in another component. It doesn't work. At least in my case it isn't. `Navigation` routers should be the registered element in `AppRegistry`. 

Although I noticed `react-redux`'s `<Provider></Provider>' can be used to wrap Navigation router's outermost component. 

Just that components from `react-native` module should not be used to wrap this. 

_Sent from my Motorola Moto G (4) using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
